app/cdap/components/common/BoxContainer/index.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import { Box } from '@material-ui/core';
+import styled from 'styled-components';
+import { IBoxContainerProps } from 'components/common/BoxContainer/types';
+
+const SimpleBox = styled(Box)`
+  display: block;
+  position: relative;
+  width: ${({ width }) => (width ? width : 'auto')};
+  height: ${({ height }) => (height ? height : 'auto')};
+  box-sizing: border-box;
+  background-color: ${({ backgroundColor }) => (backgroundColor ? backgroundColor : 'transparent')};
+  padding: ${({ padding }) => (padding ? padding : 0)};
+  margin: ${({ margin }) => (margin ? margin : 0)};
+  border-radius: ${({ borderRadius }) => (borderRadius ? borderRadius : 0)};
+  border: ${({ border }) => (border ? border : 0)};
+  text-align: ${({ textAlign }) => (textAlign ? textAlign : 'left')};
+`;
+
+export const BlockContainer = ({
+  children,
+  padding,
+  margin,
+  backgroundColor,
+  borderRadius,
+  border,
+  width,
+  height,
+  onClick,
+  dataTestId,
+  textAlign,
+}: IBoxContainerProps) => (
+  <SimpleBox
+    padding={padding}
+    margin={margin}
+    backgroundColor={backgroundColor}
+    borderRadius={borderRadius}
+    border={border}
+    width={width}
+    height={height}
+    onClick={onClick}
+    dataTestId={dataTestId}
+    textAlign={textAlign}
+  >
+    {children}
+  </SimpleBox>
+);
+
+const AbsoluteDiv = styled(SimpleBox)`
+  position: absolute;
+  top: ${({ top }) => (top ? top : 0)};
+  left: ${({ left }) => (left ? left : 0)};
+  right: ${({ right }) => (right ? right : 0)};
+  bottom: ${({ bottom }) => (bottom ? bottom : 0)};
+`;
+
+export const AbsoluteContainer = ({
+  children,
+  padding,
+  margin,
```

Comment:
why are we taking all these values in props
are all of them required for each component?

i want component stylings to be fixed and we can keep height or width dynamic for now

plz let me know if this is fine


const AbsoluteDiv1 = styled(SimpleBox)`
  position: absolute;
  top: 10px;
  left: 5px;
`;

const AbsoluteDiv2 = styled(SimpleBox)`
  position: absolute;
  top: 11px;
  left: 12px;
`;